arch/x86/kvm/pkvm/pkvm.c

```diff
@@ -578,6 +578,24 @@ static bool is_guest_vcpu_accessible(struct kvm_vcpu *vcpu)
 	return !pkvm_is_protected_vcpu(vcpu);
 }
 
+static void pkvm_update_exception_bitmap(struct kvm_vcpu *vcpu)
+{
+	/*
+	 * The guest_debug will impact what exceptions should be intercepted
+	 * for the debugging purpose. Debugging npVMs from the host side is
+	 * allowed thus updating its guest_debug flags accordingly, but
+	 * debugging pVMs from the host side is not allowed.
+	 *
+	 * As the __pkvm__update_exception_bitmap is always denied for the pVM,
+	 * it must be a code bug if the vcpu is protected.
+	 */
+	BUG_ON(pkvm_is_protected_vcpu(vcpu));
+
```

Comment:
nit: the comment above is first of all about setting `guest_debug`, and only after that about the BUG_ON. So maybe remove this empty line, to make it clear that the comment relates to the `vcpu->guest_debug = ...` line, not just to the BUG_ON?